spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -779,4 +761,56 @@ private MessageHandlerMethodFactory createDefaultMessageHandlerMethodFactory() {
 
 	}
 
+	private class TypeMetadataHolder {
```

Comment:
The idea was to minimize time while the Map segment is locked, however I agree thread contention is not that high here.